### PR TITLE
[WIP] Initial implementation of input hiding

### PR DESCRIPTION
**Pull request by ellisonbg**
This is an extremely basic implementation of code cell input hiding. Right now it is only usable from the command pallete and there is no UI indicating that the cell has hidden input, which doesn't look very good. It also does not persist the hiding to the notebook document. More work to do here, but a start.

### CODE REVIEW

**Comment by Carreau** [comment]
I'm almost wondering is input shoudl be an attribute on cell: 

`cell.input.hidden=true` Seem more readable, and guess we will have request for more attributes.

**Comment by ellisonbg** [comment]
When the input area is a separate view/model, that might make sense. But
for now I think this is sufficient.

On Mon, Oct 19, 2015 at 7:33 PM, Matthias Bussonnier <
notifications@github.com> wrote:

> I'm almost wondering is input shoudl be an attribute on cell:
>
> cell.input.hidden=true Seem more readable, and guess we will have request
> for more attributes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/pull/622#issuecomment-149376331>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

File: notebook/static/notebook/js/notebook.js [hunk 1]
```diff
@@ -1563,6 +1563,46 @@ define(function (require) {
...
+    Notebook.prototype.show_code_cell_input = function() {
+        var cells = this.get_marked_cells();
+        if (cells.length === 0) {
+            cells = [this.get_selected_cell()];
+        }
+        cells.forEach(function(cell) {cell.hide_input=false;});
+    };
+
+    /**
+     * Hide the input of the all cells
```
Comment (Jeffiy) [inline notebook/static/notebook/js/notebook.js]: `Hide the input of all cells` should be better?

File: notebook/static/notebook/js/codecell.js [hunk 2]
```diff
@@ -439,6 +439,29 @@ define([
...
+    /**
+     * Whether or not the cells input is hidden.
+     * @return {boolean}
+     */
+    Object.defineProperty(CodeCell.prototype, 'hide_input', {
+        get: function() {
+            return this.element.hasClass('hide-input');
+        },
+        set: function(value) {
+            var isHidden = this.element.hasClass('hide-input');
```
Comment (minrk) [inline notebook/static/notebook/js/codecell.js]: Presumably changes should be persisted to `this.metadata.hide_input`, and initial state loaded in fromJSON, yes?

File: notebook/static/notebook/js/notebook.js (hunk 1 above)
Comment (jdfreder) [inline notebook/static/notebook/js/notebook.js]: yes

File: notebook/static/notebook/js/codecell.js (hunk 2 above)
Comment (damianavila) [inline notebook/static/notebook/js/codecell.js]: I think so...

**Comment by Carreau** [comment]
Trying to revive this, 
and I will also push again forward the idea that the UI should be 3 state.
`hidden`/`scroll`/`expanded`
This could even be an `int`: 
  - 0: collapsed
  - N: scroll N lines.
  - inf (that we can encode as -1) expanded.

**Comment by ellisonbg** [comment]
I think the main thing that needs to happen for this to move forward is to
work with Cameron on the UI/UX design of it. All of the implementation
questions should follow from that.

On Sat, Feb 6, 2016 at 10:02 PM, Matthias Bussonnier <
notifications@github.com> wrote:

> Trying to revive this,
> and I will also push again forward the idea that the UI should be 3 state.
> hidden/scroll/expanded
> This could even be an int:
>
>    - 0: collapsed
>    - N: scroll N lines.
>    - inf (that we can encode as -1) expanded.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/pull/622#issuecomment-180958503>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by Carreau** [comment]
ping @jamiehq

**Comment by Carreau** [comment]
Oops, ping @jamiesHQ with an S

**Comment by Carreau** [comment]
Tagging as `closed-PR`, and closing as this is seem to have stalled a bit. 
This is just to keep the PR queue a bit shorter and focusing on what is actually worked on.
Feel free to reopen, (or ask for reopening if you don't have enough right to do so). 
If you were not the original author, and want to revive this, please feel free to do so. 

Also, we are doing our best to enable people to learn how to contribute to the project, and get commits in. If you are missing time or have any other issues that prevent you to work on that, please feel free to tell us, we can try to build on top of what you did. 

Thanks a lot and looking forward to see this moving forward again.

--- 

Brian, looking forward to your UI prototype with @cameronoelsen and see that revived. 

Also will need rebase.  Thanks !

**Comment by ellisonbg** [comment]
Totally fine closing this - do we have an issue to track the feature though?

On Thu, Apr 7, 2016 at 2:23 PM, Matthias Bussonnier <
notifications@github.com> wrote:

> Tagging as closed-PR, and closing as this is seem to have stalled a bit.
> This is just to keep the PR queue a bit shorter and focusing on what is
> actually worked on.
> Feel free to reopen, (or ask for reopening if you don't have enough right
> to do so).
> If you were not the original author, and want to revive this, please feel
> free to do so.
>
> Also, we are doing our best to enable people to learn how to contribute to
> the project, and get commits in. If you are missing time or have any other
> issues that prevent you to work on that, please feel free to tell us, we
> can try to build on top of what you did.
>
> Thanks a lot and looking forward to see this moving forward again.
> ------------------------------
>
> Brian, looking forward to your UI prototype with @cameronoelsen
> <https://github.com/cameronoelsen> and see that revived.
>
> Also will need rebase. Thanks !
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/pull/622#issuecomment-207093096>
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Pull request closed without merging**